dds/DCPS/ValueDispatcher.h

```diff
@@ -21,8 +21,8 @@ struct OpenDDS_Dcps_Export ValueDispatcher {
   virtual void* new_value() const = 0;
   virtual void delete_value(void* data) const = 0;
 
-  virtual bool read(ValueReader& value_reader, void* data) const = 0;
-  virtual bool write(ValueWriter& value_writer, const void* data) const = 0;
+  virtual bool read(ValueReader& value_reader, void* data, bool key_only = false) const = 0;
+  virtual bool write(ValueWriter& value_writer, const void* data, bool key_only = false) const = 0;
```

Comment:
It looks like these functions are called on topic types, so it only needs to differentiate between full sample vs. key-only sample, and a bool would be sufficient.

Reply:
A `bool` is certainly sufficient.  Bool parameters are often avoided in API design because the functional call expression lacks context about what the `true` and `false` constants mean.

Reply:
Got it. Not sure if we have an enum that only has `Full` and `KeyOnly`. If not, I think using `Sample::Extent` is okay; it would just not be called with `Sample::Extent::NestedKeyOnly`.